src/Compilers/CSharp/Test/Emit/PDB/PortablePdbTests.cs

```diff
@@ -209,6 +210,151 @@ public static void Main()
             }
         }
 
+        [Fact]
+        public void PortablePdb_DeterministicCompilation()
+        {
+            var referenceTimestamp = DateTime.Now;
+
+            var referenceCompilation = CreateCompilation(
+@"public struct StructWithReference
+{
+    string PrivateData;
+}
+public struct StructWithValue
+{
+    int PrivateData;
+}", options: TestOptions.DebugDll.WithCurrentLocalTime(referenceTimestamp));
+
+            string source = @"
+using System;
+
+class C
+{
+    public static void Main()
+    {
+        Console.WriteLine();
+    }
+}
+";
+            var originalCompilation = CreateCompilation(
+                Parse(source, "goo.cs"),
+                references: new MetadataReference[] { new CSharpCompilationReference(referenceCompilation) },
```

Comment:
>CSharpCompilationReference [](start = 58, length = 26)

As noted previously we should not worry about Compilation references and just emit info about PE references. So to emulate that you'll need to create PE reference - see EmitToArray and EmitToImageReference helpers.

Reply:
I saw those helpers but still didn't 100% understand. I'll ping you tomorrow about it.